helm-charts/charts/kube-starrocks/charts/starrocks/templates/_helpers.tpl

```diff
@@ -119,6 +119,57 @@ be.conf: |
 {{- print "/opt/starrocks/cn/log" }}
 {{- end }}
 
+{{/*
+Define a function to handle resource limits for fe
+*/}}
+{{- define "starrockscluster.fe.resources" -}}
+resources:
+  requests:
+  {{- range $key, $value := .Values.starrocksFESpec.resources.requests }}
+    {{ $key }}: {{ $value }}
+  {{- end }}
+  limits:
+  {{- range $key, $value := .Values.starrocksFESpec.resources.limits }}
+    {{- if ne (toString $value) "unlimited" }}
+    {{ $key }}: {{ $value }}
+    {{- end }}
+  {{- end }}
+{{- end -}}
+
+{{/*
+Define a function to handle resource limits for be
+*/}}
+{{- define "starrockscluster.be.resources" -}}
+resources:
+  requests:
+  {{- range $key, $value := .Values.starrocksBeSpec.resources.requests }}
+    {{ $key }}: {{ $value }}
+  {{- end }}
+  limits:
+  {{- range $key, $value := .Values.starrocksBeSpec.resources.limits }}
+    {{- if ne (toString $value) "unlimited" }}
+    {{ $key }}: {{ $value }}
+    {{- end }}
+  {{- end }}
+{{- end -}}
+
+{{/*
+Define a function to handle resource limits for cn
+*/}}
+{{- define "starrockscluster.cn.resources" -}}
+resources:
+  requests:
+  {{- range $key, $value := .Values.starrocksCnSpec.resources.requests }}
```

Comment:
why not directly `toYaml` here since there is no specific processing to the key/value.

Reply:
No special reason, just keep same with limits.
I will try to change to `toYaml`

Reply:
committed